packages/cli/core/src/commands/ui/deploy.ts

```diff
@@ -159,14 +160,47 @@ export default class Deploy extends CLICommand {
     this.validateDeployConfigJson(deployConfigPath, deployConfig);
     const hostedPage = this.buildHostedPage(deployConfig);
 
-    if (flags.pageId) {
-      this.updatePage({...hostedPage, id: flags.pageId});
+    const pageId = flags.pageId || (await this.getPageIdFromPage(hostedPage));
+
+    if (pageId) {
+      await this.updatePage({...hostedPage, id: pageId});
       return;
     }
 
     await this.createPage(hostedPage);
   }
 
+  private async getPageIdFromPage(
```

Comment:
I was confused initially when reading this function. I think it's because of the name.

It's doing more than a simple `getPageId` as its name implies.

One suggestion would be on line 163, to modify the code similar to this: 

```typescript
if(flags.pageId) {
  await this.updatePage({...hostedPage, id: pageId});
   return;
} 

await (this.possiblyOverwritePageWithDuplicateName(hostedPage) || this.createPage(hostedPage))

```

Reply:
I disagree, I think it's better for the 'frontend' code to limit the number of assertions that results in a network egress, especially for non-idempotent calls.
So essentially, I think we should first determine what kind of network call we want to do (update or create) and then have a single call to it.

I agree however that the getPageId is poorly named and could leave to confusion, I'll rework that.